src/binding/python/BaseRecordComponent.cpp

```diff
@@ -49,22 +49,30 @@ void init_BaseRecordComponent(py::module &m) {
             using DT = Datatype;
 
             // ref: https://docs.scipy.org/doc/numpy/user/basics.types.html
-            if( brc.getDatatype() == DT::CHAR )
+            if( brc.getDatatype() == DT::BOOL )
+                return py::dtype("?");
+            else if( brc.getDatatype() == DT::CHAR )
                 return py::dtype("b");
             else if( brc.getDatatype() == DT::UCHAR )
                 return py::dtype("B");
-            else if( brc.getDatatype() == DT::INT16 )
-                return py::dtype("int16");
-            else if( brc.getDatatype() == DT::INT32 )
-                return py::dtype("int32");
-            else if( brc.getDatatype() == DT::INT64 )
-                return py::dtype("int64");
-            else if( brc.getDatatype() == DT::UINT16 )
-                return py::dtype("uint16");
-            else if( brc.getDatatype() == DT::UINT32 )
-                return py::dtype("uint32");
-            else if( brc.getDatatype() == DT::UINT64 )
-                return py::dtype("uint64");
+            else if( brc.getDatatype() == DT::SHORT )
+                return py::dtype("short");
+            else if( brc.getDatatype() == DT::INT )
+                return py::dtype("int");
+            // missing in numpy: covered by uint or ulonglong
+            // else if( brc.getDatatype() == DT::LONG )
+            //     return py::dtype("long");
```

Comment:
Now this is weird.
https://docs.scipy.org/doc/numpy/user/basics.types.html
> `int_` | Default integer type (same as C `long`; normally either `int64` or `int32`)

https://docs.scipy.org/doc/numpy-1.14.2/reference/arrays.scalars.html
> `int_` | compatible: Python int | `'l'`

https://docs.python.org/3/c-api/long.html
(Note that there's two meanings of `long`here: (a) C `long` (b) Pythons's 'bignum' integer of arbitrary precision)
> PyLong_FromLong(long v)
Note the lacking `PyLong_FromInt(int v)`

So there seems to be an equivalent of `long` in `int_`, but no counterpart to `unsigned long`.

Reply:
https://docs.scipy.org/doc/numpy/user/basics.types.html#array-types-and-conversions-between-types
> Additionally to `intc` the platform dependent C integer types `short`, `long`, `longlong` and their unsigned versions are defined.

Quick test:
```Python
In [1]: import numpy as np

In [2]: np.dtype('long')
Out[2]: dtype('int64')
```

Haven't figured out how `unsigned long` works yet.

Reply:
Wow, as fancy as my float-confusion in Python xD